core/crypto/keccak.go

```diff
@@ -0,0 +1,23 @@
+package crypto
+
+import (
+	"github.com/NethermindEth/juno/core/felt"
+	"golang.org/x/crypto/sha3"
+)
+
+var h = sha3.NewLegacyKeccak256()
```

Comment:
Why are we using the legacy version?

Docs say;

```
// Only use this function if you require compatibility with an existing cryptosystem
// that uses non-standard padding. All other users should use New256 instead.
```

Reply:
Yes, because legacy keccak and sha3 use different padding, so it results in a different hash. Otherwise, they would have said to use sha3 instead of Keccak.

Reply:
Please have a look [here](https://ethereum.stackexchange.com/questions/30369/difference-between-keccak256-and-sha3). `LegacyKeccak256` is not officially supported by NIST.

Reply:
Got it, thanks